testsuite/tests/singlecluster/mtls/test_mtls_functional_behavior_with_authpolicy.py

```diff
@@ -0,0 +1,68 @@
+"""
+Tests the enabling and disabling of mTLS configuration via the Kuadrant CR (authpolicy)
+"""
+
+import time
+
+import pytest
+
+pytestmark = [pytest.mark.kuadrant_only, pytest.mark.disruptive]
+
+
+@pytest.fixture(scope="module")
+def rate_limit(cluster, blame, module_label, gateway, route):  # pylint: disable=unused-argument
+    """RateLimitPolicy for testing"""
+    return None
+
+
+def test_requests_succeed_when_mtls_disabled_auth_policy(
+    kuadrant, client, authorization, auth
+):  # pylint: disable=unused-argument
+    """Tests that requests succeed when mTLS is disabled"""
+    kuadrant.refresh().model.spec.mtls = {"enable": False}
+    kuadrant.apply()
+    kuadrant.wait_for_ready()
+
+    assert kuadrant.model.spec.mtls.enable is False
```

Comment:
I think `wait_for_ready()` on CustomResource class does not issue `refresh()` so I would call it here to check the object in cluster not the cached one in memory. But I may be mistaken